src/clj/puppetlabs/puppetserver/ringutils.clj

```diff
@@ -85,6 +109,28 @@
         false))
     true))
 
+(schema/defn ^:always-validate request->name :- (schema/maybe schema/Str)
+  "Return the name of the client by parsing the subject of the client cert from
+  the request.  Return nil if the client cert is absent or cannot be parsed."
+  [request :- RingRequest]
+  (let [{:keys [ssl-client-cert]} request]
+    (if ssl-client-cert (get-cert-subject ssl-client-cert))))
+
+(schema/defn ^:always-validate authorized? :- schema/Bool
+  "(SERVER-768) replacement for client-allowed-access?"
+  [rules :- rules/Rules
+   request :- RingRequest]
+  (if-let [name (request->name request)]
+    (let [request-for-tk-authz-schema (merge {:remote-addr ""} request)
```

Comment:
Ah, okay, makes sense.  When tests are resurrected to replace this, seems like it would be better to work this into the mocked request than to have to put this into production code.